src/main/kotlin/com/terraformation/backend/seedbank/model/AccessionHistoryModel.kt

```diff
@@ -1,17 +1,57 @@
 package com.terraformation.backend.seedbank.model
 
 import com.terraformation.backend.db.UserId
+import java.time.Instant
 import java.time.LocalDate
 
 enum class AccessionHistoryType {
   Created,
+  /** Manual edits to the accession's seed quantity, not including withdrawals. */
+  QuantityUpdated,
   StateChanged,
+  /** A withdrawal for the purpose of viability testing. */
+  ViabilityTesting,
+  /** A withdrawal for a purpose other than viability testing. */
+  Withdrawal,
 }
 
 data class AccessionHistoryModel(
+    /**
+     * When the underlying object representing the event was created. Used as a secondary sort key
+     * when multiple events happened on the same date.
+     */
+    val createdTime: Instant,
+    /**
+     * The effective date of the event. For example, if the event is a withdrawal, this would be the
+     * withdrawal date entered by the user (which might not be the same day as [createdTime].)
+     */
     val date: LocalDate,
     val description: String,
+    /**
+     * The full name of the user who was responsible for the event. This may or may not be a
+     * registered Terraware user; in some places in the app, we let people enter names of other
+     * people as plain text fields.
+     */
+    val fullName: String?,
     val type: AccessionHistoryType,
-    val userId: UserId,
-    val userName: String?,
-)
+    /** If the event was attributed to a Terraware user, that user's ID. */
+    val userId: UserId?,
+) : Comparable<AccessionHistoryModel> {
+  /**
+   * Compares two history models for sorting purposes. We sort history in reverse [date] order, with
+   * each day's history sorted in reverse [createdTime] order. We do it this way rather than just
+   * sorting by [createdTime] because some kinds of history can be backdated, e.g., on Tuesday you
+   * can tell the system that a withdrawal happened on Monday, and it should show up with Monday's
+   * date in the history records.
+   */
+  override fun compareTo(other: AccessionHistoryModel): Int {
+    // We sort history in reverse time order, so the comparison results are flipped here.
```

Comment:
Most recent event appears last?

Reply:
Most recent first, to match what's in the designs. That said: do you think it'd be better to return the results in chronological (earliest first) order and let the client display them in reverse order? I was trying to make the API match the UI here but maybe that's the wrong design choice.

Reply:
Ideally maybe .. since Ui will be the only client using this API, we could leave it as-is.

Reply:
OK, will leave it for now and we can easily revisit later if we want.